### PR TITLE
Bug 1947067: vendor/AlecAivazis/survey: Bump API to v2

### PR DESCRIPTION
**Pull request by mtnbikenc**
Updated survey.AskOne to pass survey.WithValidator option instead of function

https://github.com/AlecAivazis/survey/releases/tag/v2.0.0

### CODE REVIEW

File: pkg/asset/installconfig/aws/basedomain.go [hunk 1]
```diff
@@ -61,14 +61,16 @@ func GetBaseDomain() (string, error) {
...
+	},
+		&domain,
+		survey.WithValidator(func(ans interface{}) error {
+			choice := ans.(string)
+			i := sort.SearchStrings(publicZones, choice)
+			if i == len(publicZones) || publicZones[i] != choice {
+				return errors.Errorf("invalid base domain %q", choice)
+			}
+			return nil
+		})); err != nil {
```
Comment (staebler) [inline pkg/asset/installconfig/aws/basedomain.go:73]: This indention does not look quite right. The parameters are indented at different levels.
```suggestion
		},
		&domain,
		survey.WithValidator(func(ans interface{}) error {
			choice := ans.(string)
			i := sort.SearchStrings(publicZones, choice)
			if i == len(publicZones) || publicZones[i] != choice {
				return errors.Errorf("invalid base domain %q", choice)
			}
			return nil
		}),
	); err != nil {
```

**Review by staebler — COMMENTED** [review]
This looks good save for a couple nits.
Could you separate the vendoring changes (go.mod, go.sum, and vendor directory) into their own commit?

**Comment by mtnbikenc** [comment]
> This looks good save for a couple nits.
> Could you separate the vendoring changes (go.mod, go.sum, and vendor directory) into their own commit?

I will update the indentation and split the vendoring into it's own commit.  I had considered doing that at first, but I was also trying to maintain each commit being a buildable commit.  Since vendoring would break code I decided to keep it together.

**Comment by mtnbikenc** [comment]
@staebler I've split the changes into two commits and updated the indentation.  Please look it over and let me know if there is anything else.

**Comment by mtnbikenc** [comment]
Updated `survey.Select` questions to use `core.OptionAnswer` answer types.
/hold cancel

**Review by staebler — APPROVED** [review]
/lgtm

**Pull request merged**